### PR TITLE
multi: cleanup

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: system/core.go [hunk 1]
```diff
@@ -38,8 +38,7 @@ func GojiWebHandlerFunc(h http.HandlerFunc) web.HandlerFunc {
 
 func (application *Application) Init(APISecret string, baseURL string,
```
Comment (itswisdomagain) [inline system/core.go]: `APISecret, baseURL, cookieSecret string, cookieSecure bool, DBHost, DBName, DBPassword, DBPort, DBUser string`?

File: controllers/main.go [hunk 2]
```diff
@@ -226,6 +225,7 @@ func dcrDataAgendas(url string) ([]*dcrdatatypes.AgendasInfo, error) {
 		return nil, err
 	}
 	data, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
```
Comment (JoeGruffins) [inline controllers/main.go]: oops

File: controllers/main.go [hunk 3]
```diff
@@ -1258,10 +1270,9 @@ func (controller *MainController) PasswordResetPost(c web.C, r *http.Request) (s
 	if !controller.IsCaptchaDone(c) {
 		session.AddFlash("You must complete the captcha.", "passwordresetError")
 		return controller.PasswordReset(c, r)
-	} else {
-		session.Values["CaptchaDone"] = false
-		c.Env["CaptchaDone"] = false
 	}
+	session.Values["CaptchaDone"] = false
```
Comment (JoeGruffins) [inline controllers/main.go]: @amassari also noticed this in #491

File: controllers/main.go [hunk 4]
```diff
@@ -1210,16 +1214,24 @@ func (controller *MainController) Index(c web.C, r *http.Request) (string, int)
 	t := controller.GetTemplate(c)
 
 	// execute the named template with data in c.Env
-	widgets := helpers.Parse(t, "home", c.Env)
-
+	widgets, err := helpers.Parse(t, "home", c.Env)
+	if err != nil {
+		log.Infof("helpers.Parse: home failed: %v", err)
+		return "/error?r=/stats", http.StatusSeeOther
+	}
```
Comment (JoeGruffins) [inline controllers/main.go]: This changes behavior, but it looks reasonable.

File: server.go [hunk 5]
```diff
@@ -276,7 +276,11 @@ func runMain() error {
 	graceful.PostHook(func() {
 		application.Close()
 	})
-	app.Abandon(middleware.Logger)
+	err = app.Abandon(middleware.Logger)
+	if err != nil {
+		return fmt.Errorf("could not Abandon logger: %v", err)
+	}
+
```
Comment (JoeGruffins) [inline server.go]: This also changes behavior

File: controllers/main.go [hunk 6]
```diff
@@ -1210,16 +1214,24 @@ func (controller *MainController) Index(c web.C, r *http.Request) (string, int)
 	t := controller.GetTemplate(c)
 
 	// execute the named template with data in c.Env
-	widgets := helpers.Parse(t, "home", c.Env)
-
+	widgets, err := helpers.Parse(t, "home", c.Env)
+	if err != nil {
+		log.Infof("helpers.Parse: home failed: %v", err)
```
Comment (jholdstock) [inline controllers/main.go]: Log error here instead of info

File: controllers/main.go [hunk 7]
```diff
@@ -1210,16 +1214,24 @@ func (controller *MainController) Index(c web.C, r *http.Request) (string, int)
...
+		return "/error?r=/stats", http.StatusSeeOther
```
Comment (jholdstock) [inline controllers/main.go]: We removed the functionality to redirect from the error page. Param `?r=/stats` is not used anywhere and can be removed.

I notice this is present in a few other places, not related to your PR. Would you mind removing them please?

File: controllers/main.go [hunk 8]
```diff
@@ -1210,16 +1214,24 @@ func (controller *MainController) Index(c web.C, r *http.Request) (string, int)
...
 	c.Env["IsIndex"] = true
 	c.Env["Title"] = "Decred Voting Service - Welcome"
 	c.Env["Designation"] = controller.Cfg.Designation
 
 	c.Env["Content"] = template.HTML(widgets)
 
-	return helpers.Parse(t, "main", c.Env), http.StatusOK
+	doc, err := helpers.Parse(t, "main", c.Env)
+	if err != nil {
+		log.Infof("helpers.Parse: main failed: %v", err)
```
Comment (jholdstock) [inline controllers/main.go]: Same 2 comments as above - remove `?r=` param and log level error

File: controllers/main.go (hunk 4 above)
Comment (jholdstock) [reply controllers/main.go]: Agreed, tested this out and the new behaviour is an improvement.

**Pull request merged**